cap_client/cap_api.py

```diff
@@ -206,13 +206,16 @@ def delete(self, pid=None):
                            method='delete',
                            expected_status_code=204)
 
-    def update(self, pid=None, filename=''):
+    def update(self, pid=None, json_=''):
         """Update an analysis by given pid and JSON data from file."""
         try:
-            data = json.loads(filename)
+            data = json.loads(json_)
         except ValueError:
-            with open(filename) as fp:
-                data = json.load(fp)
+            try:
+                with open(json_) as fp:
+                    data = json.load(fp)
+            except (IOError, ValueError):
+                raise MissingJsonFile()
 
         json_data = json.dumps(data)
 
```

Comment:
you can see that returned response is different than in all other client's methods. It's because you didn't specify a serializer, so client is getting all our internal data from the server.
You need to pass `'Accept': 'application/basic+json'` in headers for `_make_request` method.